report/renderer/details-renderer.js

```diff
@@ -376,7 +379,116 @@ export class DetailsRenderer {
   }
 
   /**
-   * @param {{headings: TableColumnHeading[], items: TableItem[]}} details
+   * Grab a matching entity from classification by its name
+   * @param {string} name
+   * @return {LH.Result.LhrEntity|undefined}
+   */
+  _getEntityByName(name) {
+    if (!this._entities) return;
+    const entityIndex = this._entities.entityIndexByName[name];
```

Comment:
Did we ever resolve the discussion on this?

I still think that if we need to map entities by name, the map should be constructed in the report renderer and not shipped as a part of the LHR.

Reply:
This is a fresh angle and implementation you're bringing up :). We're relying on the entity classification format we shipped earlier. 

FWIW, I don't see an advantage to removing lookup tables and reconstructing it during reporting. I'm assuming you're suggesting to pack origins as a property of each entity, but that makes the structure unusable without post processing (for us and other JSON consumers). What's the downside you see in having lookup tables?

Reply:
The difference is wasted computation cycles each time we have to access or do something with the structure. Neither structures are friendly to human consumption of this data, so I've traded to optimize on performance and converting O(n) lookups to O(1) via lookup tables, while minimizing redundancy of information. Sure the perf in terms of a single LightHouse manual run is trivial, but it could make a difference in repeated runs if anyone's doing the report at scale in the cloud.

Reply:
Okay, so I'll start making those changes as a separate PR and rebase this against that.

Reply:
https://github.com/GoogleChrome/lighthouse/pull/14744